src/services/message.ts

```diff
@@ -105,6 +105,48 @@ export class MessageService implements threema.MessageService {
     }
 
     /**
+     * return the filename of a message (image, audio, file)
+     * used for downloads
+     * @param message
+     * @returns filename string or null
+     */
+    public getFileName(message: threema.Message): string {
+        if (message === undefined) {
+            return null;
+        }
+
+        let getFileName = (prefix: string, postfix: string = null): string => {
+            return prefix
+                + '-' + message.id
+                + (postfix !== null ? '.' + postfix : '');
```

Comment:
Instead of using an explicit `null` default value, you could also make the postfix optional, in which case it'll be `undefined`:

```typescript
let getFileName = (prefix: string, postfix?: string): string => {
    // ...
    + (postfix !== undefined ? '.' + postfix : '');
};
```